path_pubkey.go

```diff
@@ -0,0 +1,97 @@
+package gcpkms
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/errwrap"
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+
+	kmspb "google.golang.org/genproto/googleapis/cloud/kms/v1"
+)
+
+func (b *backend) pathPubkey() *framework.Path {
+	return &framework.Path{
+		Pattern: "pubkey/" + framework.GenericNameRegex("key"),
+
+		HelpSynopsis: "Retrieve the public key associated with the named key",
+		HelpDescription: `
+Retrieve the PEM-encoded Google Cloud KMS public key associated with the Vault
+named key. The key will only be available if the key is asymmetric.
+`,
+
+		Fields: map[string]*framework.FieldSchema{
+			"key": &framework.FieldSchema{
+				Type: framework.TypeString,
+				Description: `
+Name of the key for which to get the public key. This key must already exist in
+Vault and Google Cloud KMS.
+`,
+			},
+
+			"key_version": &framework.FieldSchema{
+				Type: framework.TypeInt,
+				Description: `
+Integer version of the crypto key version from which to exact the public key.
+This field is required.
+`,
+			},
+		},
+
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.ReadOperation: withFieldValidator(b.pathPubkeyRead),
```

Comment:
Should this be an Update operation instead since you're requiring `key_version`?

Reply:
Hmm I actually think it should be GET. It's possible to have params for get requests (they turn into URL params). I'm going to fix the curl command because `vault write gcpkms/pubkey/my-key key_verison=1` doesn't feel right.

Reply:
I agree that read feels correct, though as is now the `key_version` would be a query parameter which is uncommon in Vault (AFAIK), and a little odd with it being required. Maybe it could be a path element, which is a pattern that would match: https://www.vaultproject.io/api/secret/transit/index.html#export-key

@briankassouf Might have some thoughts on the API conventions here.

Reply:
As in `vault read gcpkms/pubkey/my-key/1`? I'm not opposed to that, but can @briankassouf weight in quickly before I make that change?

Reply:
Using a GET parameter is fine, it's not a common pattern because it wasn't part of our framework implementation until recently. The kv version 2 API uses get parameters to retrieve a specific version of a secret. This aligns with that same usecase.

Reply:
Hey @sethvargo one last question. The KVV2 API defaults to the latest key version if `version` isn't provided. Does that concept make sense here? I briefly skimmed the Google docs but it wasn't totally clear to me.

Reply:
I thought about it, but it would involve an extra API call (we'd have to list the key versions from the KMS api and grab the latest one). Two problems arise:

1. What is "latest"? Is it the latest active key or the most recently created key?
1. What if the user doesn't have permission to list all the keys.

That's why I opted to keep it required.